lib/shared/bucketing/__tests__/bucketing.test.ts

```diff
@@ -1415,6 +1415,124 @@ describe('Rollout Logic', () => {
         })
+
+        it('should handle stepped rollout with 0% start and 100% later stage', () => {
+            const rollout: PublicRollout = {
+                type: 'stepped',
+                startDate: new Date(
+                    new Date().getTime() - 1000 * 60 * 60 * 24 * 7,
+                ),
+                startPercentage: 0,
+                stages: [
+                    {
+                        type: 'discrete',
+                        date: new Date(
+                            new Date().getTime() + 1000 * 60 * 60 * 24 * 7,
+                        ),
+                        percentage: 1,
+                    },
+                ],
+            }
+
+            // Before next stage - should be 0%
+            jest.useFakeTimers().setSystemTime(new Date())
+            for (let i = 0; i < 100; i++) {
+                expect(
+                    doesUserPassRollout({ rollout, boundedHash: i / 100 }),
+                ).toBeFalsy()
+            }
+
+            // After 100% stage - should pass all users
+            jest.useFakeTimers().setSystemTime(
+                new Date(new Date().getTime() + 1000 * 60 * 60 * 24 * 8),
+            )
+            for (let i = 0; i < 100; i++) {
+                expect(
+                    doesUserPassRollout({ rollout, boundedHash: i / 100 }),
+                ).toBeTruthy()
+            }
+
+            jest.useRealTimers()
+        })
+
+        it.only('should handle stepped rollout with 50% start and 100% later stage', () => {
+            const rollout: PublicRollout = {
+                type: 'stepped',
+                startDate: new Date(
+                    new Date().getTime() - 1000 * 60 * 60 * 24 * 7,
+                ),
+                startPercentage: 0.5,
+                stages: [
+                    {
+                        type: 'discrete',
+                        date: new Date(
+                            new Date().getTime() + 1000 * 60 * 60 * 24 * 7,
+                        ),
+                        percentage: 1,
+                    },
+                ],
+            }
+
+            // Before next stage - should be 50%
+            jest.useFakeTimers().setSystemTime(new Date())
+            expect(
+                doesUserPassRollout({ rollout, boundedHash: 0.49 }),
+            ).toBeTruthy()
+            expect(
+                doesUserPassRollout({ rollout, boundedHash: 0.51 }),
+            ).toBeFalsy()
+
+            // After 100% stage - should pass all users
+            jest.useFakeTimers().setSystemTime(
+                new Date(new Date().getTime() + 1000 * 60 * 60 * 24 * 8),
+            )
+            for (let i = 0; i < 100; i++) {
+                expect(
+                    doesUserPassRollout({ rollout, boundedHash: i / 100 }),
+                ).toBeTruthy()
+            }
+
+            jest.useRealTimers()
+        })
+
+        it('should handle stepped rollout with 100% start and 0% later stage', () => {
```

Comment:
can you add a test case for when the starting step is in the future, to make sure it's zero